cores/esp8266/PolledTimeout.h

```diff
@@ -89,14 +173,24 @@ class timeoutTemplate
   
   bool checkExpired(const timeType t) const
   {
-    return (t - _start) >=  _timeout;
+    // always returns false when _timeout is ::max()
+    return (t - _start) > _timeout;
```

Comment:
I understand why you did this, but it means the timeout calc is now off by 1 unit, e.g. if you construct with timeout 10 it won't timeout at 10, but at 11.

Reply:
How about this:
```cpp
(t - _start) % TimePolicyT::timeMax() > _timeout;
```

Reply:
Note for future ref: the ESP8266 does not have a hw div instruction, hence operator % is expensive. Instead, pay the price of an extra flag in memory.